pkg/googlecloud/googlecloud.go

```diff
@@ -0,0 +1,104 @@
+package googlecloud
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"bigtable-autoscaler.com/m/v2/pkg/interfaces"
+	"cloud.google.com/go/bigtable"
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+type googleCloudClient struct {
+	metricsClient  interfaces.MetricClientWrapper
+	bigtableClient interfaces.BigtableClientWrapper
+	projectID      string
+	instanceID     string
+	clusterID      string
+	ctx            context.Context
+}
+
+func NewClient(ctx context.Context, credentialsJSON []byte, projectID, instanceID, clusterID string) (interfaces.GoogleCloudClient, error) {
+	metricClient, err := monitoring.NewMetricClient(ctx, option.WithCredentialsJSON(credentialsJSON))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create metrics client: %w", err)
+	}
+	metricClientWrapped := metricClientWrapper{
+		metricsClient: metricClient,
+	}
+
+	bigtableClient, err := bigtable.NewInstanceAdminClient(ctx, projectID, option.WithCredentialsJSON(credentialsJSON))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create bigtable client: %w", err)
+	}
+	bigtableClientWrapped := bigtableClientWrapper{
+		bigtableClient: bigtableClient,
+	}
+
+	return &googleCloudClient{
+		metricsClient:  &metricClientWrapped,
+		bigtableClient: &bigtableClientWrapped,
+		projectID:      projectID,
+		instanceID:     instanceID,
+		clusterID:      clusterID,
+		ctx:            ctx,
+	}, nil
+}
+
+func (m *googleCloudClient) GetLastCPUMeasure() (int32, error) {
```

Comment:
Maybe this is a better name:
```suggestion
func (m *googleCloudClient) GetCurrentCPULoad() (int32, error) {
```